sdk/keyvault/internal/challenge_policy_test.go

```diff
@@ -38,12 +38,37 @@ func TestFindScopeAndTenant(t *testing.T) {
 	}
 
 	resp.Header.Set("WWW-Authenticate", "Bearer authorization=\"https://login.microsoftonline.com/00000000-0000-0000-0000-000000000000\", resource=\"https://managedhsm.azure.net\" scope=\"https://vault.azure.net/.default\"")
+	p.findScopeAndTenant(&resp)
+	if *p.scope != scope {
+		t.Fatalf("scope was not properly parsed, got %s, expected %s", *p.scope, scope)
+	}
+	if *p.tenantID != fakeTenant {
+		t.Fatalf("tenant ID was not properly parsed, got %s, expected %s", *p.tenantID, fakeTenant)
+	}
 
+	resp.Header.Set("WWW-Authenticate", "Bearer resource=\"https://managedhsm.azure.net\" scope=\"https://vault.azure.net/.default\", authorization=\"https://login.microsoftonline.com/00000000-0000-0000-0000-000000000000\"")
```

Comment:
Suggest building challenges with e.g. `fmt.Sprintf` to make clear which parameter has the expected value